src/Components/Server/src/Circuits/DefaultCircuitFactory.cs

```diff
@@ -64,15 +77,46 @@ public override CircuitHost CreateCircuitHost(HttpContext httpContext, IClientPr
                 client,
                 rendererRegistry,
                 renderer,
-                config,
+                components,
+                dispatcher,
                 jsRuntime,
                 circuitHandlers);
 
-            // Initialize per-circuit data that services need
+            // Initialize per - circuit data that services need
+#pragma warning disable CS0618 // Type or member is obsolete
             (circuitHost.Services.GetRequiredService<IJSRuntimeAccessor>() as DefaultJSRuntimeAccessor).JSRuntime = jsRuntime;
+#pragma warning restore CS0618 // Type or member is obsolete
             (circuitHost.Services.GetRequiredService<ICircuitAccessor>() as DefaultCircuitAccessor).Circuit = circuitHost.Circuit;
 
             return circuitHost;
         }
+
+        private static IList<ComponentDescriptor> ResolveComponentMetadata(HttpContext httpContext, IClientProxy client)
+        {
+            if (client != null)
+            {
+                var endpointFeature = httpContext.Features.Get<IEndpointFeature>();
+                var endpoint = endpointFeature?.Endpoint;
+                if (endpoint == null)
+                {
+                    throw new InvalidOperationException("CompnentHub doesn't have an associated endpoint.");
```

Comment:
Typo in `CompnentHub`

Reply:
Also if there's any chance that customers could see this error, can you make it clearer what's gone wrong? Customers won't know how to fix their code based on this message.

Reply:
I've improved the comment